lib/setUserInfo.js

```diff
@@ -0,0 +1,64 @@
+import Sequelize from 'sequelize'
+import moment from 'moment'
+import getToken from './getToken'
+import models from '../models'
+import {ADMIN_USERGROUP_ID} from '../config'
+
+export default async function setUserInfo(req) {
+	if (req.get('Authorization')) {
+		let userToken = getToken(req)
+
+		let userData
+		try {
+			userData = await models.User.findOne({
+				where: {
+					authToken: userToken
+				},
+				include: [
+					{
+						model: models.UserGroupMembership,
+						as: 'user',
+						include: [
+							{
+								model: models.UserGroup,
+								as: 'userGroup'
+							}
+						]
+					}
+				]
+			})
+		} catch (err) {
+			console.log('Error getting current user from database: ' + err)
+		}
+
+		if (!userData) {
+			return req
+		}
+
+		// simplify list of user's groups
+		userData.isAdmin = false
+		let userGroups = []
+		userData.user.forEach((ug) => {
+			userGroups.push({
+				id: ug.dataValues.userGroup.dataValues.id,
```

Comment:
These are ugly, but so is the object that sequelize gives us from the above query.